src/shims/fs.rs

```diff
@@ -440,19 +497,63 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     }
 }
 
-// Extracts the number of seconds and nanoseconds elapsed between `time` and the unix epoch, and
-// then sets the `mask` bits determined by `flag` when `time` is Ok. If `time` is an error, it
-// returns `(0, 0)` without setting any bits.
-fn extract_sec_and_nsec<'tcx>(
-    time: std::io::Result<SystemTime>,
-    mask: &mut u32,
-    flag: u32,
-) -> InterpResult<'tcx, (u64, u32)> {
-    if let Ok(time) = time {
+// Extracts the number of seconds and nanoseconds elapsed between `time` and the unix epoch when
+// `time` is Ok. If `time` is an error, it returns `None`.
```

Comment:
When writing doc comments, make sure you actually use doc comment syntax. :)

Also, the comment makes it sound like the function is infallible. When can this return `Err`?

Reply:
It can fail if the argument is a system time before the unix epoch.

Reply:
I see. Please add that to the comment.